File: Rasha_Toine/Rasha/QLForm/src/ql/ast/Block.java
package ql.ast;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

import ql.ast.statement.Statement;
import ql.visiting.StatementVisitor;


public class Block extends AstNode {

	private List<Statement> statements;

	public Block(List<Statement> statements) {
		this.statements = statements;
	}

	public Block() {
		this.statements = new ArrayList<>();
	}

    public void insertStatement(Statement statement){
        this.statements.add(statement);
    }
    
    public List<Statement> getStatements(){
        return Collections.unmodifiableList(statements);
    }
    
	public <T, U> T accept(StatementVisitor<T, U> visitor, U ctx){
		return visitor.visit(this, ctx);
    }
}

File: DogeQL/src/main/kotlin/ql/ast/node/Block.kt
package ql.ast.node

import ql.ast.location.SourceLocation
import ql.ast.node.statement.Statement
import ql.visitor.QuestionnaireASTBaseVisitor

class Block(val statements: List<Statement>, location: SourceLocation) : QLNode(location) {

    override fun <T> accept(visitor: QuestionnaireASTBaseVisitor<T>): T {
        return visitor.visit(this)
    }

}